staging/src/k8s.io/client-go/rest/config.go

```diff
@@ -672,3 +675,34 @@ func CopyConfig(config *Config) *Config {
 	}
 	return c
 }
+
+func DefaultCodecFactoryForScheme(scheme *runtime.Scheme) serializer.CodecFactory {
+	// TODO: Thoughts about this:
+
+	// This depends on a client-go feature gate, so it can't live in apimachinery. It's a
+	// substitute for conventional package-scoped variables (i.e. `var Codecs =
+	// NewCodecFactory(Scheme)`) when the caller doesn't really want to be responsible for codec
+	// manufacturing but they need a NegotiatedSerializer for their client's content config. Is
+	// there a better home for it?
+
+	// Is it worth caching these? Is there a realistic use case for creating clients so rapidly
+	// that the cost of NewCodecFactory becomes significant?
```

Comment:
Maybe? https://pkg.go.dev/golang.org/x/sync/singleflight might work here?  Its not obvious to me just eyeballing this if it's needed or helpful.

Reply:
> Maybe? https://pkg.go.dev/golang.org/x/sync/singleflight might work here? Its not obvious to me just eyeballing this if it's needed or helpful.

Perhaps a refinement before going beta.  @benluddy I suggest opening a beta requirement PR to the KEP so we dont' lose it. New clients are created for some use-cases around "now connect with these credentials".